src/main/kotlin/com/dietmap/yaak/domain/googleplay/GooglePlaySubscriptionService.kt

```diff
@@ -117,12 +103,29 @@ class GooglePlaySubscriptionService(val androidPublisherApiClient: AndroidPublis
                 price = BigDecimal(subscription.priceAmountMicros).divide(BigDecimal(1000 * 1000)),
                 currencyCode = subscription.priceCurrencyCode,
                 transactionId = subscription.orderId,
-                originalTransactionId = toInitialOrderId(subscription.orderId),
+                originalTransactionId = subscription.getInitialOrderId(),
                 appMarketplace = AppMarketplace.GOOGLE_PLAY,
                 expiryTimeMillis = subscription.expiryTimeMillis,
                 googlePlayPurchaseDetails = GooglePlayPurchaseDetails(packageName, notification.subscriptionId, notification.purchaseToken)
         )
         userAppClient.sendSubscriptionNotification(subscriptionUpdate)
         logger.info { "Google Play subscription notification has been sent to user app: $subscriptionUpdate" }
     }
+
+    private fun SubscriptionPurchase.calculateEffectivePrice() = when {
```

Comment:
Could you please verify it with the Android team? I think they are sending in some cases `effectivePrice` in order request.

Reply:
According to the Android team it's still the same as before - either `priceAmountMicros` or `introductoryPriceInfo.introductoryPriceAmountMicros`. I've just extracted the method and replaced some `if`s with `when`, but the logic is the same.